### PR TITLE
Model Updates

### PR DESCRIPTION
**Pull request by daviddyess**
- Added missing model for user_profile
- Added 'has' associations to Recipe model
- Added 'has' associations to User model
- Added 'has' associations to Flavor model
- Added 'has' associations to Vendor model
- Added 'has' associations to DataSupplier model
- Added 'has' associations to Diluent model

### CODE REVIEW

**Comment by daviddyess** [comment]
What started as just adding a model, turned into adding has... associations to models as well.

File: src/models/diluent.js [hunk 1]
```diff
@@ -32,8 +32,10 @@ module.exports = (sequelize, DataTypes) => {
...
   );
 
-  /* Diluent.associate = function(models) {
-    // associations can be defined here
-  };*/
+  Diluent.associate = function(models) {
+    this.hasMany(models.RecipesDiluents, {
+      foreignKey: 'diluentId'
+    });
+  };
```
Comment (ayan4m1) [inline src/models/diluent.js]: There's also a hasMany relationship to PreparationsDiluents, right?

File: src/models/flavor.js [hunk 2]
```diff
@@ -39,6 +39,9 @@ module.exports = (sequelize, DataTypes) => {
 
   Flavor.associate = function(models) {
     this.belongsTo(models.Vendor, { foreignKey: 'vendorId' });
+    this.hasOne(models.FlavorIdentifier, { foreignKey: 'flavorId' });
```
Comment (ayan4m1) [inline src/models/flavor.js]: A flavor can have many flavor identifiers (from different data suppliers).

File: src/models/user.js [hunk 3]
```diff
@@ -32,8 +32,26 @@ module.exports = (sequelize, DataTypes) => {
...
+    this.hasMany(models.UsersFlavors, {
+      foreignKey: 'userId'
+    });
+    this.hasMany(models.UsersRoles, {
+      foreignKey: 'userId'
+    });
+    this.hasOne(models.UserProfile, {
+      foreignKey: 'userId'
+    });
+    this.hasOne(models.UserToken, {
```
Comment (ayan4m1) [inline src/models/user.js]: There can be many tokens for a given user. If they are logged in on multiple devices, multiple browsers, if they close a browser window and then log in again - plenty of cases where it's expected.

File: src/models/vendor.js [hunk 4]
```diff
@@ -29,8 +29,9 @@ module.exports = (sequelize, DataTypes) => {
     }
   );
 
-  Vendor.associate = function() {
-    //
+  Vendor.associate = function(models) {
+    this.hasOne(models.VendorIdentifier, { foreignKey: 'flavorId' });
```
Comment (ayan4m1) [inline src/models/vendor.js]: A vendor can have multiple identifiers, up to one per data supplier.

**Comment by ayan4m1** [comment]
Looks pretty good - I think a few of the associations need to be updated (left comments) but other than that this is sorely needed.

File: src/models/diluent.js (hunk 1 above)
Comment (daviddyess) [reply src/models/diluent.js]: I'm not seeing a table preparations_diluents, was that added recently?

File: src/models/user.js (hunk 3 above)
Comment (daviddyess) [reply src/models/user.js]: Fixed

File: src/models/flavor.js (hunk 2 above)
Comment (daviddyess) [reply src/models/flavor.js]: Fixed

File: src/models/vendor.js (hunk 4 above)
Comment (daviddyess) [reply src/models/vendor.js]: Fixed

File: src/models/diluent.js (hunk 1 above)
Comment (ayan4m1) [reply src/models/diluent.js]: Good catch, added in #11 !

File: src/models/userProfile.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        type: DataTypes.STRING,
+        allowNull: true
+      },
+      bio: {
+        type: DataTypes.TEXT,
+        allowNull: true
+      },
+      url: {
+        type: DataTypes.STRING,
+        allowNul: true
```
Comment (ayan4m1) [inline src/models/userProfile.js]: I think this should be `allowNull`

**Pull request merged**